search/search-dev-tools/search-dev-tools.php

```diff
@@ -0,0 +1,185 @@
+<?php
+// phpcs:disable WordPress.PHP.DisallowShortTernary.Found
+namespace Automattic\VIP\Search\Dev_Tools;
+
+use Automattic\VIP\Search\Search;
+
+if ( ! ( defined( 'VIP_ENABLE_VIP_SEARCH' ) && VIP_ENABLE_VIP_SEARCH ) ) {
+	return;
+}
+
+define( 'SEARCH_DEV_TOOLS_CAP', 'edit_others_posts' );
+
+add_action(
+	'rest_api_init',
+	function() {
+		register_rest_route(
+			'vip/v1',
+			'search/repl',
+			[
+				'methods'             => [
+					'POST',
+				],
+				'callback'            => __NAMESPACE__ . '\rest_callback',
+				'permission_callback' => __NAMESPACE__ . '\should_enable_search_dev_tools',
+				// Uncomment this for testing with `npm run dev`
+				// 'permission_callback' => '__return_true',
+				'args'                => [
+					'url'   => [
+						'type'              => 'string',
+						'required'          => true,
+						'validate_callback' => function( $value, $request, $param ) {
+								return filter_var( $value, FILTER_VALIDATE_URL ) ?: new \WP_Error( 'rest_invalid_param', sprintf( '%s is not a valid URL', $param ) );
+						},
+					],
+					'query' => [
+						'type'              => 'string',
+						'required'          => true,
+						'validate_callback' => function( $value, $request, $param ) {
+							json_decode( $value );
+							return JSON_ERROR_NONE === json_last_error() ?: new \WP_Error( 'rest_invalid_param', sprintf( '%s is not a valid JSON', $param ) );
+						},
+					],
+				],
+			]
+		);
+	}
+);
+
+/**
+ * REST API wrapper to query the ES
+ *
+ * @param \WP_REST_Request $request
+ * @return void
+ */
+function rest_callback( \WP_REST_Request $request ) {
+
+	$ep             = \ElasticPress\Elasticsearch::factory();
+	$result         = $ep->remote_request(
+		trim( parse_url( $request['url'], PHP_URL_PATH ), '/' ),
+		[
+			'body'   => $request['query'],
+			'method' => 'POST',
+		]
+	);
+	$result['body'] = json_encode( json_decode( $result['body'] ), JSON_PRETTY_PRINT );
+	return rest_ensure_response( [ 'result' => $result ] );
+}
+
+
+/**
+ * A capability-based check for whether Dev Tools should be enabled or not
+ *
+ * @return boolean
+ */
+function should_enable_search_dev_tools(): bool {
+	return current_user_can( SEARCH_DEV_TOOLS_CAP );
+}
+
+function is_ratelimited(): bool {
+	return wp_cache_get( Search::QUERY_COUNT_CACHE_KEY, Search::QUERY_COUNT_CACHE_GROUP ) > Search::$max_query_count;
+}
+
+add_action( 'wp_enqueue_scripts', function() {
+	if ( ! should_enable_search_dev_tools() ) {
+		return;
+	}
+
+	// @todo: remove cachebusting arg.
+	wp_enqueue_script( 'vip-search-dev-tools', plugin_dir_url( __FILE__ ) . 'build/bundle.js', [], current_time( 'timestamp' ), true );
+	wp_enqueue_style( 'vip-search-dev-tools', plugin_dir_url( __FILE__ ) . 'build/bundle.css', [], current_time( 'timestamp' ) );
+}, 11 );
+
+add_action(
+	'wp_body_open',
```

Comment:
Seeing as though this just outputs a `script` element and a `div`, could the PHP array not be provided to the front-end via `wp_localize_script()` in the above function, and that leaves just the `div`, which would be much easier to document as "If you have the hook, this is added for you. If you don't, add this `div` to your `header.php`."

It would then also avoid having to handle the escaping of the PHP values (of various types) into JavaScript in the right way (`wp_json_encode()` doesn't make it safe).

Reply:
This hook is not going to be used in production, it was a quick test thing. The mount point is going to be rendered on `wp_footer`